sync/syncer.go

```diff
@@ -308,18 +328,18 @@ func (s *Syncer) synchronise() {
 	}
 
 	// node is not synced
-	s.handleNotSynced(s.lValidator.ProcessedLayer() + 1)
+	s.handleNotSynced(s.ProcessedLayer() + 1)
 }
 
 //validate all layers except current one
 func (s *Syncer) handleLayersTillCurrent() {
 	//dont handle current
-	if s.lValidator.ProcessedLayer()+1 >= s.GetCurrentLayer() {
+	if s.ProcessedLayer()+1 >= s.GetCurrentLayer() {
 		return
 	}
 
-	s.Info("handle layers %d to %d", s.lValidator.ProcessedLayer()+1, s.GetCurrentLayer()-1)
-	for currentSyncLayer := s.lValidator.ProcessedLayer() + 1; currentSyncLayer < s.GetCurrentLayer(); currentSyncLayer++ {
+	s.Info("handle layers %d to %d", s.ProcessedLayer()+1, s.GetCurrentLayer()-1)
+	for currentSyncLayer := s.ProcessedLayer() + 1; currentSyncLayer < s.GetCurrentLayer(); currentSyncLayer++ {
```

Comment:
What is the case where we may have more than one unprocessed layer? Only if we have a large clock drift and we 'skip' a layer? What happens if one of the layers in the for loop is empty? Is this case possible?

Reply:
its possible the flow should behave the same